src/app/dim-api/actions.ts

```diff
@@ -62,44 +66,49 @@ const installApiPermissionObserver = _.once(() => {
  */
 const installObservers = _.once((dispatch: ThunkDispatch<RootState, undefined, AnyAction>) => {
   // Watch the state and write it out to IndexedDB
-  observeStore(
-    (state) => state.dimApi,
-    _.debounce((currentState: DimApiState, nextState: DimApiState) => {
-      if (
-        // Avoid writing back what we just loaded from IDB
-        currentState?.profileLoadedFromIndexedDb &&
-        // Check to make sure one of the fields we care about has changed
-        (nextState.settings !== currentState.settings ||
-          nextState.profiles !== currentState.profiles ||
-          nextState.updateQueue !== currentState.updateQueue ||
-          nextState.itemHashTags !== currentState.itemHashTags ||
-          nextState.searches !== currentState.searches)
-      ) {
-        // Only save the difference between the current and default settings
-        const settingsToSave = subtractObject(nextState.settings, initialSettingsState) as Settings;
-
-        const savedState: ProfileIndexedDBState = {
-          settings: settingsToSave,
-          profiles: nextState.profiles,
-          updateQueue: nextState.updateQueue,
-          itemHashTags: nextState.itemHashTags,
-          searches: nextState.searches,
-        };
-        infoLog('dim sync', 'Saving profile data to IDB');
-        set('dim-api-profile', savedState);
-      }
-    }, 1000),
-  );
+  observe({
+    id: 'profile-observer',
+    getObserved: (state) => state.dimApi,
+    sideEffect: _.debounce(
```

Comment:
@bhollis when would you expect these side effects to be run? I am doing a bit of testing and having trouble triggering this one and the one below.

Looking at the code, I would expect both of these to be triggered by a loadout update as per this https://github.com/DestinyItemManager/DIM/blob/dim-api-observers/src/app/dim-api/reducer.ts#L866-L891

Reply:
Figured it out, forgot to wrap the `observe` invocation in a `dispatch`.